Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift

```diff
@@ -586,6 +589,23 @@ internal class MiniAppScriptMessageHandler: NSObject, WKScriptMessageHandler {
         }
     }
 
+    private func purchaseProduct(with callbackId: String, parameters: RequestParameters?) {
+        if let purchaseItemId = parameters?.itemId {
+            hostAppMessageDelegate?.purchaseProduct(withId: purchaseItemId, completionHandler: { result in
+                switch result {
+                case .success(let response):
+                    guard let encodedResponse = ResponseEncoder.encode(data: response) else {
+                        self.executeJavaScriptCallback(responseStatus: .onError, messageId: callbackId, response: getMiniAppErrorMessage(MiniAppJavaScriptError.internalError))
+                        return
+                    }
+                    self.executeJavaScriptCallback(responseStatus: .onSuccess, messageId: callbackId, response: encodedResponse)
+                case .failure(let error):
+                    self.executeJavaScriptCallback(responseStatus: .onError, messageId: callbackId, response: getMiniAppErrorMessage(error))
```

Comment:
Maybe we should return an error object instead of just a String? Like we did with the access token and some other features.

I'm just thinking in the future there could be many different error cases that need to be handled. In general, it seems better to switch to using error objects instead of strings for all new features.

Reply:
Done